### PR TITLE
Datadog reporter

### PR DESCRIPTION
**Pull request by mgobec**
* Updated ReporterConfig so that the options are more flexible
* Added datadog UDP reporter with tests

### CODE REVIEW

File: cassandra-diagnostics-reporter-datadog/src/main/java/io/smartcat/cassandra/diagnostics/reporter/DatadogReporter.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     * Constructor.
+     *
+     * @param configuration Reporter configuration
+     */
+    public DatadogReporter(ReporterConfiguration configuration) {
+        super(configuration);
+
+        logger.debug("Initializing datadog client with config: {}", configuration.toString());
+
+        hostname = getHostname();
```
Comment (nbozic-zz) [inline cassandra-diagnostics-reporter-datadog/src/main/java/io/smartcat/cassandra/diagnostics/reporter/DatadogReporter.java:52]: Think each and every reporter will have hostname, IMO all logic to pull hostname should go to `Reporter` class and should be done same way for all reporters. All measurements should not have hostname defined in it but glued by reporter on single place.

File: cassandra-diagnostics-reporter-datadog/src/main/java/io/smartcat/cassandra/diagnostics/reporter/DatadogReporter.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            client.recordGaugeValue(measurement.name(), measurement.value(), convertTagsMap(measurement.tags()));
+            logger.info("Reporting measurement {}, value {} and tags {}", measurement.name(), measurement.value(),
+                    convertTagsMap(measurement.tags()));
+        } catch (Exception e) {
+            logger.warn("Sending measurement failed: execTime={}, exception: {}", measurement.time(), e.getMessage());
+        }
+    }
+
+    private String getHostname() {
+        try {
```
Comment (nbozic-zz) [inline cassandra-diagnostics-reporter-datadog/src/main/java/io/smartcat/cassandra/diagnostics/reporter/DatadogReporter.java:87]: We should move this code to Reporter, and do `super.getHostname()` or something like this.

**Review by nbozic-zz — COMMENTED** [review]
Not scope of this PR but Kafka test is failing:
`KafkaReporterTest.send_measurements:82 expected:<1> but was:<0>`

**Comment by mgobec** [comment]
Looks like that getCanonicalHostName() gets a fully qualified hostname and we can use that. The questions is should this be a part of reporter code or module that generates measurements. I would propose to have a hostname helper in commons and use as needed.

**Comment by nbozic-zz** [comment]
I do not see any measurement useful without hostname, so it looks to me it should be at either core or reporter. Reporters are mostly binding hostname if it is not defined, so they will need hostname handling and it looks to me it is right place to put it.

**Comment by nivancevic** [comment]
The problem with getCanonicalHostName() and FQ hostname is twofold: 
1) it could triggers expensive DNS resolution process upon call
2) metrics collected by telegraf (or some other metrics agent) usually use hostname (that could differ from FQ hostname determined by interface IP address)

We could solve 1) with proper caching. In order to solve 2) (and 1) partially) I would suggest we introduce a new configuration property `hostname` that eliminates the need for 1) and can be used to cope with 2).

**Comment by nivancevic** [comment]
Moreover we could use the configuration to instruct diagnostics modules/reporters to skip adding the hostname tag as it could be done elsewhere (e.g. shipping agent like telegraf, riemann server, etc.).

**Comment by mgobec** [comment]
Merging this work and working on #136 in a separate PR

**Pull request merged**